src/bin.ts

```diff
@@ -220,6 +222,44 @@ const deploy = async ({ yes, bucket }: { yes: boolean, bucket: string }) => {
             }
         });
 
+        promises.push(...permanentRedirects.map(async redirect => {
+            const { fromPath, toPath } = redirect
+
+            const key = withoutLeadingSlash(fromPath)
+            const redirectLocation = '/' + withoutTrailingSlash(withoutLeadingSlash(toPath))
+            const tag = `"${createHash('md5').update(redirectLocation).digest('hex')}"`;
+            const object = objects.find(object => object.Key === key && object.ETag === tag);
+
+            isKeyInUse[key] = true;
+
+            if (object) {
+                // object with exact hash already exists, abort.
+                return;
+            }
+
+            try {
+                const promise = new S3.ManagedUpload({
+                    service: s3,
+                    params: {
+                        Bucket: config.bucketName,
+                        Key: key,
+                        Body: toPath,
```

Comment:
I think this should be redirectLocation instead of toPath, because the md5 ETag we check is based on redirectPath.

Alternatively could we just upload an empty object here (`Buffer.alloc(0)`) and compare using WebsiteRedirectLocation instead of the ETag? We're most likely going to stop using ETag's anyway due to #59.

Reply:
Sounds good, I'll use an empty object here.  The ETag is probably still a good idea, because it seems that WebsiteRedirectLocation is not readily available from the `listObjectsV2` API call.

![Screen Shot 2019-05-08 at 9 37 52 AM](https://user-images.githubusercontent.com/430257/57383808-0f1a3980-7175-11e9-935c-e0aaab53e404.png)

Reply:
If we upload empty contents, ETag would be the same for all redirects which means if a redirect changes we won't know to update it. Maybe we can just use ETag for now and figure it out once we work out what we're doing with #59. Alternatively, uploading a 0B file wouldn't take long so we could just PUT redirects regardless.

Reply:
Ah, I see, I for some reason thought that we were assigning the ETag in the upload request.